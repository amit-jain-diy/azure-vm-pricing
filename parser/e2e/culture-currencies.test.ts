const child_process = require('child_process');
const fs = require('fs');
const readline = require('readline');

describe('End-to-end tests for supported cultures and currencies', () => {
  describe('en-us - English (US)', () => {
    const culture = 'en-us';

    test('USD - US Dollar ($)', (done) => {
      const currency = 'USD';

      assert(done, culture, currency);
    });

    test('SAR - Saudi Riyal (SR)', (done) => {
      const currency = 'SAR';

      assert(done, culture, currency);
    });

    test('IDR - Indonesian Rupiah (Rp)', (done) => {
      const currency = 'IDR';

      assert(done, culture, currency);
    });
  });

  describe('cs-cz - Čeština', () => {
    const culture = 'cs-cz';

    test('EUR - Euro (€)', (done) => {
      const currency = 'EUR';

      assert(done, culture, currency);
    });
  });

  describe('da-dk - Dansk', () => {
    const culture = 'da-dk';

    test('DKK - Danish Krone (kr)', (done) => {
      const currency = 'DKK';

      assert(done, culture, currency);
    });
  });

  describe('de-de - Deutsch', () => {
    const culture = 'de-de';

    test('EUR - Euro (€)', (done) => {
      const currency = 'EUR';

      assert(done, culture, currency);
    });

    test('CHF - Swiss Franc. (chf)', (done) => {
      const currency = 'CHF';

      assert(done, culture, currency);
    });
  });

  describe('en-au - English (Australia)', () => {
    const culture = 'en-au';

    test('AUD - Australian Dollar ($)', (done) => {
      const currency = 'AUD';

      assert(done, culture, currency);
    });
  });

  describe('en-ca - English (Canada)', () => {
    const culture = 'en-ca';

    test('CAD - Canadian Dollar ($)', (done) => {
      const currency = 'CAD';

      assert(done, culture, currency);
    });
  });

  describe('en-in - English (India)', () => {
    const culture = 'en-in';

    test('INR - Indian Rupee (₹)', (done) => {
      const currency = 'INR';

      assert(done, culture, currency);
    });
  });

  describe('en-gb - English (UK)', () => {
    const culture = 'en-gb';

    test('GBP - British Pound (£)', (done) => {
      const currency = 'GBP';

      assert(done, culture, currency);
    });

    test('MYR - Malaysian Ringgit (RM$)', (done) => {
      const currency = 'MYR';

      assert(done, culture, currency);
    });

    test('ZAR - South African Rand (R)', (done) => {
      const currency = 'ZAR';

      assert(done, culture, currency);
    });

    test('NZD - New Zealand Dollar ($)', (done) => {
      const currency = 'NZD';

      assert(done, culture, currency);
    });

    test('HKD - Hong Kong Dollar (HK$)', (done) => {
      const currency = 'HKD';

      assert(done, culture, currency);
    });
  });

  describe('es-es - Español', () => {
    const culture = 'es-es';

    test('EUR - Euro (€)', (done) => {
      const currency = 'EUR';

      assert(done, culture, currency);
    });

    test('ARS - Argentine Peso ($)', (done) => {
      const currency = 'ARS';

      assert(done, culture, currency);
    });
  });

  describe('es-mx - Español (MX)', () => {
    const culture = 'es-mx';

    test('MXN - Mexican Peso (MXN$)', (done) => {
      const currency = 'MXN';

      assert(done, culture, currency);
    });
  });

  describe('fr-fr - Français', () => {
    const culture = 'fr-fr';

    test('EUR - Euro (€)', (done) => {
      const currency = 'EUR';

      assert(done, culture, currency);
    });

    test('CHF - Swiss Franc. (chf)', (done) => {
      const currency = 'CHF';

      assert(done, culture, currency);
    });
  });

  describe('fr-ca - Français (Canada)', () => {
    const culture = 'fr-ca';

    test('CAD - Canadian Dollar ($)', (done) => {
      const currency = 'CAD';

      assert(done, culture, currency);
    });
  });

  describe('it-it - Italiano', () => {
    const culture = 'it-it';

    test('EUR - Euro (€)', (done) => {
      const currency = 'EUR';

      assert(done, culture, currency);
    });

    test('CHF - Swiss Franc. (chf)', (done) => {
      const currency = 'CHF';

      assert(done, culture, currency);
    });
  });

  describe('hu-hu - Magyar', () => {
    const culture = 'hu-hu';

    test('EUR - Euro (€)', (done) => {
      const currency = 'EUR';

      assert(done, culture, currency);
    });
  });

  describe('nb-no - Norsk', () => {
    const culture = 'nb-no';

    test('NOK - Norwegian Krone (kr)', (done) => {
      const currency = 'NOK';

      assert(done, culture, currency);
    });
  });

  describe('nl-nl - Nederlands', () => {
    const culture = 'nl-nl';

    test('EUR - Euro (€)', (done) => {
      const currency = 'EUR';

      assert(done, culture, currency);
    });
  });

  describe('pl-pl - Polski', () => {
    const culture = 'pl-pl';

    test('EUR - Euro (€)', (done) => {
      const currency = 'EUR';

      assert(done, culture, currency);
    });
  });

  describe('pt-br - Português (Brasil)', () => {
    const culture = 'pt-br';

    test('BRL - Brazilian Real (R$)', (done) => {
      const currency = 'BRL';

      assert(done, culture, currency);
    });
  });

  describe('pt-pt - Português', () => {
    const culture = 'pt-pt';

    test('EUR - Euro (€)', (done) => {
      const currency = 'EUR';

      assert(done, culture, currency);
    });
  });

  describe('sv-se - Svenska', () => {
    const culture = 'sv-se';

    test('SEK - Swedish Krona (kr)', (done) => {
      const currency = 'SEK';

      assert(done, culture, currency);
    });
  });

  describe('tr-tr - Türkçe', () => {
    const culture = 'tr-tr';

    test('TRY - Turkish Lira (TL)', (done) => {
      const currency = 'TRY';

      assert(done, culture, currency);
    });
  });

  describe('ru-ru - Pусский', () => {
    const culture = 'ru-ru';

    test('RUB - Russian Ruble (руб)', (done) => {
      const currency = 'RUB';

      assert(done, culture, currency);
    });
  });

  describe('ja-jp - 日本語', () => {
    const culture = 'ja-jp';

    test('JPY - Japanese Yen (¥)', (done) => {
      const currency = 'JPY';

      assert(done, culture, currency);
    });
  });

  describe('ko-kr - 한국어', () => {
    const culture = 'ko-kr';

    test('KRW - Korean Won (₩)', (done) => {
      const currency = 'KRW';

      assert(done, culture, currency);
    });
  });

  describe('zh-tw - 中文(繁體)', () => {
    const culture = 'zh-tw';

    test('TWD - Taiwanese Dollar (NT$)', (done) => {
      const currency = 'TWD';

      assert(done, culture, currency);
    });

    test('HKD - Hong Kong Dollar (HK$)', (done) => {
      const currency = 'HKD';

      assert(done, culture, currency);
    });
  });
});

function assert(
  done: jest.DoneCallback,
  culture: string,
  currency: string
): void {
  const operatingSystem = 'windows';
  const region = 'us-west';

  const crawler = child_process.spawn(
    'yarn',
    [
      'crawl',
      '-l',
      culture,
      '-c',
      currency,
      '-o',
      operatingSystem,
      '-r',
      region
    ],
    { shell: true }
  );

  let crawlerErrors = [];

  crawler.stderr.on('data', (data) => {
    crawlerErrors.push(data);
  });

  crawler.on('close', (code) => {
    expect(code).toBe(0);

    if (crawlerErrors.length > 0) {
      done.fail(`Errors: ${crawlerErrors}`);
    }

    const fileStream = fs.createReadStream(
      `./out/vm-pricing_${region}_${operatingSystem}.csv`
    );
    const rl = readline.createInterface({
      input: fileStream,
      crlfDelay: Infinity
    });

    let foundExpectedVirtualMachine = false;

    rl.on('close', () => {
      if (!foundExpectedVirtualMachine) {
        throw 'Did not find "D2 v3" virtual machine.';
      }
    });

    rl.on('line', (line) => {
      if (line.startsWith('D2 v3,')) {
        foundExpectedVirtualMachine = true;
        rl.close();
        expect(line).toMatchSnapshot();
        done();
      }
    });
  });
}
