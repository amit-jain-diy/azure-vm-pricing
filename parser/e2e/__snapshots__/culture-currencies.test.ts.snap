// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End-to-end tests for supported cultures and currencies cs-cz - Čeština EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,1763\\",\\"0,0987\\",\\"0,1445\\",\\"0,067\\",\\"0,1204\\",\\"0,0429\\",\\"0,0717\\",\\"0,0287\\""`;

exports[`End-to-end tests for supported cultures and currencies da-dk - Dansk DKK - Danish Krone (kr) 1`] = `"D2 v3,2,8,\\"1,3154\\",\\"0,7364\\",\\"1,0784\\",\\"0,4994\\",\\"0,8985\\",\\"0,3195\\",\\"0,535\\",\\"0,214\\""`;

exports[`End-to-end tests for supported cultures and currencies de-de - Deutsch CHF - Swiss Franc. (chf) 1`] = `"D2 v3,2,8,\\"0,2057\\",\\"0,1152\\",\\"0,1687\\",\\"0,0781\\",\\"0,1405\\",\\"0,05\\",\\"0,0837\\",\\"0,0335\\""`;

exports[`End-to-end tests for supported cultures and currencies de-de - Deutsch EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,1763\\",\\"0,0987\\",\\"0,1445\\",\\"0,067\\",\\"0,1204\\",\\"0,0429\\",\\"0,0717\\",\\"0,0287\\""`;

exports[`End-to-end tests for supported cultures and currencies en-au - English (Australia) AUD - Australian Dollar ($) 1`] = `"D2 v3,2,8,0.287,0.1607,0.2353,0.109,0.1961,0.0697,0.1168,0.0467"`;

exports[`End-to-end tests for supported cultures and currencies en-ca - English (Canada) CAD - Canadian Dollar ($) 1`] = `"D2 v3,2,8,0.2676,0.1498,0.2194,0.1016,0.1828,0.065,0.1088,0.0436"`;

exports[`End-to-end tests for supported cultures and currencies en-gb - English (UK) GBP - British Pound (£) 1`] = `"D2 v3,2,8,0.1558,0.0873,0.1278,0.0592,0.1065,0.0379,0.0634,0.0254"`;

exports[`End-to-end tests for supported cultures and currencies en-gb - English (UK) HKD - Hong Kong Dollar (HK$) 1`] = `"D2 v3,2,8,1.6219,0.908,1.3296,0.6157,1.1079,0.3939,0.6596,0.2638"`;

exports[`End-to-end tests for supported cultures and currencies en-gb - English (UK) MYR - Malaysian Ringgit (RM$) 1`] = `"D2 v3,2,8,0.8778,0.4914,0.7197,0.3333,0.5996,0.2132,0.357,0.1428"`;

exports[`End-to-end tests for supported cultures and currencies en-gb - English (UK) NZD - New Zealand Dollar ($) 1`] = `"D2 v3,2,8,3.831,2.1447,3.1407,1.4544,2.6168,0.9305,1.5581,0.6231"`;

exports[`End-to-end tests for supported cultures and currencies en-gb - English (UK) NZD - New Zealand Dollar ($) 2`] = `"D2 v3,2,8,0.316,0.1769,0.2591,0.12,0.2159,0.0768,0.1286,0.0514"`;

exports[`End-to-end tests for supported cultures and currencies en-gb - English (UK) ZAR - South African Rand (R) 1`] = `"D2 v3,2,8,3.831,2.1447,3.1407,1.4544,2.6168,0.9305,1.5581,0.6231"`;

exports[`End-to-end tests for supported cultures and currencies en-in - English (India) INR - Indian Rupee (₹) 1`] = `"D2 v3,2,8,15.0575,8.4293,12.3443,5.7161,10.2852,3.6571,6.1239,2.4487"`;

exports[`End-to-end tests for supported cultures and currencies en-us - English (US) IDR - Indonesian Rupiah (Rp) 1`] = `"D2 v3,2,8,3268.76,1829.88,2679.7576,1240.8776,2232.7664,793.8864,1329.4,531.5724"`;

exports[`End-to-end tests for supported cultures and currencies en-us - English (US) SAR - Saudi Riyal (SR) 1`] = `"D2 v3,2,8,3268.76,1829.88,2679.7576,1240.8776,2232.7664,793.8864,1303.25,521.1561"`;

exports[`End-to-end tests for supported cultures and currencies en-us - English (US) SAR - Saudi Riyal (SR) 2`] = `"D2 v3,2,8,0.7838,0.4388,0.6426,0.2976,0.5354,0.1904,0.3188,0.1275"`;

exports[`End-to-end tests for supported cultures and currencies en-us - English (US) USD - US Dollar ($) 1`] = `"D2 v3,2,8,0.209,0.117,0.1714,0.0794,0.1428,0.0508,0.085,0.034"`;

exports[`End-to-end tests for supported cultures and currencies es-es - Español ARS - Argentine Peso ($) 1`] = `"D2 v3,2,8,\\"17,4578\\",\\"9,7731\\",\\"14,3121\\",\\"6,6273\\",\\"11,9248\\",\\"4,24\\",\\"7,1001\\",\\"2,8391\\""`;

exports[`End-to-end tests for supported cultures and currencies es-es - Español EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,1763\\",\\"0,0987\\",\\"0,1445\\",\\"0,067\\",\\"0,1204\\",\\"0,0429\\",\\"0,0717\\",\\"0,0287\\""`;

exports[`End-to-end tests for supported cultures and currencies es-mx - Español (MX) MXN - Mexican Peso (MXN$) 1`] = `"D2 v3,2,8,4.4099,2.4687,3.6153,1.6741,3.0123,1.0711,1.7935,0.7172"`;

exports[`End-to-end tests for supported cultures and currencies fr-ca - Français (Canada) CAD - Canadian Dollar ($) 1`] = `"D2 v3,2,8,\\"0,2676\\",\\"0,1498\\",\\"0,2194\\",\\"0,1016\\",\\"0,1828\\",\\"0,065\\",\\"0,1088\\",\\"0,0436\\""`;

exports[`End-to-end tests for supported cultures and currencies fr-fr - Français CHF - Swiss Franc. (chf) 1`] = `"D2 v3,2,8,\\"0,2057\\",\\"0,1152\\",\\"0,1687\\",\\"0,0781\\",\\"0,1405\\",\\"0,05\\",\\"0,0837\\",\\"0,0335\\""`;

exports[`End-to-end tests for supported cultures and currencies fr-fr - Français EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,1763\\",\\"0,0987\\",\\"0,1445\\",\\"0,067\\",\\"0,1204\\",\\"0,0429\\",\\"0,0717\\",\\"0,0287\\""`;

exports[`End-to-end tests for supported cultures and currencies hu-hu - Magyar EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,2057\\",\\"0,1152\\",\\"0,1687\\",\\"0,0781\\",\\"0,1405\\",\\"0,05\\",\\"0,0837\\",\\"0,0335\\""`;

exports[`End-to-end tests for supported cultures and currencies hu-hu - Magyar EUR - Euro (€) 2`] = `"D2 v3,2,8,\\"0,1763\\",\\"0,0987\\",\\"0,1445\\",\\"0,067\\",\\"0,1204\\",\\"0,0429\\",\\"0,0717\\",\\"0,0287\\""`;

exports[`End-to-end tests for supported cultures and currencies it-it - Italiano CHF - Swiss Franc. (chf) 1`] = `"D2 v3,2,8,\\"0,2057\\",\\"0,1152\\",\\"0,1687\\",\\"0,0781\\",\\"0,1405\\",\\"0,05\\",\\"0,0837\\",\\"0,0335\\""`;

exports[`End-to-end tests for supported cultures and currencies it-it - Italiano EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,1763\\",\\"0,0987\\",\\"0,1445\\",\\"0,067\\",\\"0,1204\\",\\"0,0429\\",\\"0,0717\\",\\"0,0287\\""`;

exports[`End-to-end tests for supported cultures and currencies ja-jp - 日本語 JPY - Japanese Yen (¥) 1`] = `"D2 v3,2,8,23.408,13.104,19.1901,8.8861,15.9892,5.6852,9.52,3.8067"`;

exports[`End-to-end tests for supported cultures and currencies ko-kr - 한국어 KRW - Korean Won (₩) 1`] = `"D2 v3,2,8,235.0519,131.5841,192.6976,89.2298,160.5551,57.0873,95.5953,38.2247"`;

exports[`End-to-end tests for supported cultures and currencies nb-no - Norsk NOK - Norwegian Krone (kr) 1`] = `"D2 v3,2,8,\\"1,8424\\",\\"1,0314\\",\\"1,5105\\",\\"0,6995\\",\\"1,2585\\",\\"0,4475\\",\\"0,7493\\",\\"0,2997\\""`;

exports[`End-to-end tests for supported cultures and currencies nl-nl - Nederlands EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,1763\\",\\"0,0987\\",\\"0,1445\\",\\"0,067\\",\\"0,1204\\",\\"0,0429\\",\\"0,0717\\",\\"0,0287\\""`;

exports[`End-to-end tests for supported cultures and currencies pl-pl - Polski EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,1763\\",\\"0,0987\\",\\"0,1445\\",\\"0,067\\",\\"0,1204\\",\\"0,0429\\",\\"0,0717\\",\\"0,0287\\""`;

exports[`End-to-end tests for supported cultures and currencies pt-br - Português (Brasil) BRL - Brazilian Real (R$) 1`] = `"D2 v3,2,8,\\"1,0239\\",\\"0,5732\\",\\"0,8394\\",\\"0,3887\\",\\"0,6994\\",\\"0,2487\\",\\"0,4165\\",\\"0,1666\\""`;

exports[`End-to-end tests for supported cultures and currencies pt-pt - Português EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,1763\\",\\"0,0987\\",\\"0,1445\\",\\"0,067\\",\\"0,1204\\",\\"0,0429\\",\\"0,0717\\",\\"0,0287\\""`;

exports[`End-to-end tests for supported cultures and currencies ru-ru - Pусский RUB - Russian Ruble (руб) 1`] = `"D2 v3,2,8,\\"13,0625\\",\\"7,3125\\",\\"10,7088\\",\\"4,9588\\",\\"8,9225\\",\\"3,1725\\",\\"5,3125\\",\\"2,1243\\""`;

exports[`End-to-end tests for supported cultures and currencies sv-se - Svenska SEK - Swedish Krona (kr) 1`] = `"D2 v3,2,8,\\"1,8261\\",\\"1,0223\\",\\"1,4971\\",\\"0,6932\\",\\"1,2473\\",\\"0,4435\\",\\"0,7427\\",\\"0,297\\""`;

exports[`End-to-end tests for supported cultures and currencies tr-tr - Türkçe TRY - Turkish Lira (TL) 1`] = `"D2 v3,2,8,\\"1,7757\\",\\"0,9941\\",\\"1,4558\\",\\"0,6741\\",\\"1,2129\\",\\"0,4313\\",\\"0,7222\\",\\"0,2888\\""`;

exports[`End-to-end tests for supported cultures and currencies zh-tw - 中文(繁體) HKD - Hong Kong Dollar (HK$) 1`] = `"D2 v3,2,8,1.6219,0.908,1.3296,0.6157,1.1079,0.3939,0.6596,0.2638"`;

exports[`End-to-end tests for supported cultures and currencies zh-tw - 中文(繁體) TWD - Taiwanese Dollar (NT$) 1`] = `"D2 v3,2,8,6.2814,3.5164,5.1495,2.3846,4.2906,1.5256,2.5547,1.0215"`;
